packages/framework/src/Model/Transport/TransportPriceRepository.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrameworkBundle\Model\Transport;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Shopsys\FrameworkBundle\Model\Transport\Exception\TransportPriceNotFoundException;
+
+class TransportPriceRepository
+{
+    /**
+     * @param \Doctrine\ORM\EntityManagerInterface $em
+     */
+    public function __construct(
+        protected readonly EntityManagerInterface $em,
+    ) {
+    }
+
+    /**
+     * @param int $domainId
+     * @param \Shopsys\FrameworkBundle\Model\Transport\Transport $transport
+     * @param int $cartTotalWeight
+     * @return \Shopsys\FrameworkBundle\Model\Transport\TransportPrice
+     */
+    public function getTransportPriceOnDomainByTransportAndClosestWeight(
+        int $domainId,
+        Transport $transport,
+        int $cartTotalWeight,
+    ): TransportPrice {
+        $transportPrice = $this->em->createQueryBuilder()
+            ->select('tp')
+            ->from(TransportPrice::class, 'tp')
+            ->where('tp.transport = :transport')
+            ->andWhere('tp.domainId = :domainId')
+            ->andWhere('((tp.maxWeight >= :cartTotalWeight) OR (tp.maxWeight IS NULL))')
+            ->setParameter('transport', $transport)
+            ->setParameter('domainId', $domainId)
+            ->setParameter('cartTotalWeight', $cartTotalWeight)
+            ->orderBy('tp.maxWeight', 'ASC')
```

Comment:
how are the nulls sorted?

Reply:
NULLs are the last which is convenient as it stands for "unlimited". If I wanted the Nulls to be first, I would have to use our custom `SortableNullsWalker` but this is not the case